### PR TITLE
 Issue #2811: Update the views Grid style to use the CSS grid property

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop/backdrop-issues/issues/2811
Replaces https://github.com/backdrop/backdrop/pull/1988

### CODE REVIEW

**Comment by jenlampton** [comment]
When I created a brand new view using the wizard, set the style to grid, and saved it immediately (rather than save + configure), it first rendered the output in an HTML table. I flushed all caches and then it rendered in a CSS grid. Is this expected? (see Test 2: http://2447.backdrop.backdrop.qa.backdropcms.org/test2 -- this was just created and caches have not yet been cleared)

File: core/modules/views/plugins/views_plugin_style_grid.inc [hunk 1]
```diff
@@ -38,34 +38,58 @@ class views_plugin_style_grid extends views_plugin_style {
...
       '#options' => array('horizontal' => t('Horizontal'), 'vertical' => t('Vertical')),
       '#default_value' => $this->options['alignment'],
-      '#description' => t('Horizontal alignment will place items starting in the upper left and moving right. Vertical alignment will place items starting in the upper left and moving down.'),
+    );
+    $form['alignment']['horizontal']['#description'] = t('Places items starting in the upper left and moving right.');
+    $form['alignment']['vertical']['#description'] = t('Places items starting in the upper left and moving down.');
+
+    $form['deprecated_table'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Use table (deprecated)'),
```
Comment (jenlampton) [inline core/modules/views/plugins/views_plugin_style_grid.inc]: Can we say  "Use HTML table" to be a little clearer here?

File: core/modules/views/plugins/views_plugin_style_grid.inc [hunk 2]
```diff
@@ -38,34 +38,58 @@ class views_plugin_style_grid extends views_plugin_style {
...
+      '#description' => t('Some sites might have made style changes based on the deprecated grid based on table. Only select this if absolutely needed.'),
```
Comment (jenlampton) [inline core/modules/views/plugins/views_plugin_style_grid.inc]: I'm also not sure about this description text, it seems quite verbose....  How about:
`For sites with views grids that need to use the deprecated HTML tables.`

**Comment by herbdool** [comment]
It shouldn't be switching just when clearing cache. Something is wrong

File: core/modules/views/plugins/views_plugin_style_grid.inc (hunk 1 above)
Comment (docwilmot) [reply core/modules/views/plugins/views_plugin_style_grid.inc]: Fixed

File: core/modules/views/plugins/views_plugin_style_grid.inc (hunk 2 above)
Comment (docwilmot) [reply core/modules/views/plugins/views_plugin_style_grid.inc]: Fixed

**Comment by docwilmot** [comment]
Fixed the rendering issue. The option definition for `deprecated_table` wasnt set.

**Comment by jenlampton** [comment]
awesome, testing again!

**Pull request merged**